hack/migrate-to-antrea.sh

```diff
@@ -0,0 +1,223 @@
+        --back-up                   Back up the original CNI configuration files and network policies.
+        --output-dir <DIR>          Output directory for the backup files.
+        --kubeconfig <KUBECONFIG>   Path to kubeconfig file.
+        --calico-manifest <PATH>    Path to the Calico manifest file, if not specified, the latest manifest will be used.
+        --flannel-manifest <PATH>   Path to the Flannel manifest file, if not specified, the latest manifest will be used.
+        --antrea-manifest <PATH>    Path to the Antrea manifest file, if not specified, the latest manifest will be used.
+"
+
+_calico_warning_message="
+WARNING: You are trying to migrate from Calico to Antrea. Please NOTE:
+If you are using Kubernetes origin Networkpolicy, all the network policies will still work well
+after migration.
+If you are using Calico NetworkPolicy or Calico GlobalNetworkPolicy, only PART of
+the network policies could be converted to Antrea NetworkPolicy and Antrea ClusterNetworkPolicy.
+Please check the migration document for details."
+
+function print_usage {
+    echo "$_usage"
+}
+
+FROM=""
+NETWORK_POLICY=false
+DRY_RUN=false
+BACK_UP=false
+OUTPUT_DIR=""
+KUBECONFIG="$HOME/.kube/config"
+PLATFORM=$(uname | tr '[:upper:]' '[:lower:]')
+CALICO_MANIFEST="https://docs.projectcalico.org/manifests/calico.yaml"
+FLANNEL_MANIFEST="https://raw.githubusercontent.com/coreos/flannel/master/Documentation/kube-flannel.yml"
+ANTREA_MANIFEST="./build/yamls/antrea.yml"
+
+while [[ $# -gt 0 ]]; do
+    key="$1"
+
+    case $key in
+    --from)
+        FROM="$2"
+        shift 2
+        ;;
+    --networkpolicy)
+        NETWORK_POLICY=true
+        shift
+        ;;
+    --dry-run)
+        DRY_RUN=true
+        shift
+        ;;
+    --back-up)
+        BACK_UP=true
+        shift
+        ;;
+    --output-dir)
+        OUTPUT_DIR="$2"
+        shift 2
+        ;;
+    --kubeconfig)
+        KUBECONFIG="$2"
+        shift 2
+        ;;
+    -h | --help)
+        print_usage
+        exit 0
+        ;;
+    *) # unknown option
+        echo "Unknown option $1"
+        exit 1
+        ;;
+    esac
+done
+
+if [ "$FROM" == "calico" ]; then
+    echo "$_calico_warning_message"
+    # input to continue
+    echo "========== Input 'yes' to continue =========="
+    read -r input
+    if [ "$input" != "yes" ]; then
+        echo "========== Aborting =========="
+        exit 1
+    fi
+fi
+echo "========== Building antctl =========="
+make antctl
+mv ./bin/antctl-${PLATFORM} ./bin/antctl
+
+echo "========== Checking the environment =========="
+if [ $DRY_RUN == true ]; then
+    echo "./bin/antctl migrate check-env --from $FROM"
+else
+    ./bin/antctl migrate check-env --from $FROM
+    if [ $? -ne 0 ]; then
+        echo "Failed to check the environment"
+        exit 1
+    fi
+fi
+
+if [ "$backup" == true ]; then
+    if [ "$OUTPUT_DIR" == "" ]; then
+        echo "Please specify the output directory for the backup files"
```

Comment:
Sure, using `$(pwd)` as the default value now.

Reply:
If the value is empty, we can use the default one. You can add a warning message if user set the option like `--output-dir=`. But if it's the case, you need to distinguish and print the warning message only when user is setting the option like `--output-dir=`.
If the user doesn't specify the option at all, then no warning messages.
This may also be applicable to other options like `--**-manifest`